distribution/server-dist/pom.xml

```diff
@@ -333,6 +333,7 @@
                 </dependency>
             </dependencies>
             <build>
+                <finalName>${product.name}-${product.filename.version}</finalName>
```

Comment:
Even if you build RH-SSO, the `server-dist` module contains project name and version. This resolved it.